workspaces/liferay-testray-workspace/client-extensions/liferay-testray-custom-element/src/components/ManagementToolbar/ManagementToolbarFilter.tsx

```diff
@@ -269,68 +243,69 @@ const FilterBody: React.FC<FilterBodyProps> = ({
 				>
 					{i18n.translate('clear')}
 				</ClayButton>
-			</div>
+			</ClayDropDown.Section>
 		</div>
 	);
 };
 
-const MENU_POPOVER_HEIGHT = 580;
-
 const ManagementToolbarFilter: React.FC<ManagementToolbarFilterProps> = ({
 	applyFilters = true,
 	fieldOptions = {},
 	filterSchema,
 	isLoading = false,
 }) => {
+	const buttonRef = useRef<HTMLButtonElement | null>(null);
+
 	const [visible, setVisible] = useState(false);
-	const ref = useRef<HTMLButtonElement>(null);
 
-	const [position, setPosition] = useState<number>(MENU_POPOVER_HEIGHT);
+	const handleExpand = (
+		event: React.MouseEvent<HTMLButtonElement, MouseEvent>
+	) => {
+		buttonRef.current = event.target as HTMLButtonElement;
 
-	const popoverAlignPosition =
-		position < MENU_POPOVER_HEIGHT ? 'top-right' : 'bottom-right';
+		setVisible(!visible);
```

Comment:
```suggestion
		setIsVisible(isVisible => !isVisible);
```
As react processes several setStates in one update, this can 'scramble' the toggle, as you need to take the previous value and change it. They talk a bit about this here [https://legacy.reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous](url). I also suggested isVisible, since it's a boolean.